extensions/amp-form/amp-form.md

```diff
@@ -75,7 +75,108 @@ __required__
+
+```html
+<form ... on="submit-success:success-lightbox;submit-error:error-lightbox" ...>
+</form>
+```
+
+See the [full example here](https://github.com/ampproject/amphtml/blob/master/examples/forms.amp.html).
+
+## Success/Error Response Rendering
+`amp-form` allows publishers to render the responses using [Extended Templates](https://github.com/ampproject/amphtml/blob/master/spec/amp-html-format.md#extended-templates).
+
+Using `submit-success` and `submit-error` special marker attributes, publishers can mark any **child element of form** and include a `<template></template>` tag inside it to render the response in it.
+
+The response is expected to be a valid JSON Object. For example, if the publisher's `action-xhr` endpoint returns the following responses:
+
+**Success Response**
+```json
+{
+  "name": "Jane Miller",
+  "interests": [{"name": "Basketball"}, {"name": "Swimming"}, {"name": "Reading"}],
+  "email": "email@example.com"
+}
+```
+
+**Error Response**
+```json
+{
+  "name": "Jane Miller",
+  "message": "The email (email@example.com) you used is already subscribed."
+}
+```
+
+Publishers can render these in a template inside their forms as follow.
+
+```html
+<form ...>
+    <fieldset>
+      ...
+    </fieldset>
+    <div submit-success>
+        <template type="amp-mustache">
+            Success! Thanks {{name}} for subscribing! Please make sure to check your email {{email}}
+            to confirm! After that we'll start sending you weekly articles on {{#interests}}<b>{{name}}</b> {{{{/interests}}.
+        </template>
+    </div>
+    <div submit-error>
+        <template type="amp-mustache">
+            Oops! {{name}}, {{message}}.
+        </template>
+    </div>
+</form>
+```
+
+See the [full example here](https://github.com/ampproject/amphtml/blob/master/examples/forms.amp.html).
+
+## Polyfills
+`amp-form` polyfill behaviors and functionality missing from some browsers or being implemented in the next version of CSS.
+
+#### Invalid Submit Blocking and Validation Message Bubble
+Browsers that uses webkit-based engines, currently (as of August 2016) do not support invalid form submissions. These include Safari on all platforms, and all iOS browsers. `amp-form` polyfill this behavior to block any invalid submissions and show validation message bubbles on invalid inputs.
+
+**Note**: Messages are sometimes limited to few words like "required field", these messages are provided by the browser implementation. We'll be working on allowing publisher-provided custom validation messages as well as custom validation UIs (instead of the builtin/polyfill'd bubbles).
+
+#### :user-invalid/:user-valid
+These pseudo classes are part of [future CSS Selectors 4 spec](https://drafts.csswg.org/selectors-4/#user-pseudos) and are introduced to allow better hooks for styling invalid/valid fields based on few criteria.
```

Comment:
I linked to the specs which includes a better description for it and I mentioned below the user interaction criteria.